s2s/time.py

```diff
@@ -417,38 +421,106 @@ def __repr__(self):
-        """Return indices for training data indices using given strategy."""
+    def get_cv_groups(self):
+        """Group intervals into bins.
+
+        Group intervals into bins for each anchor year by labelling.
+        args:
+            intervals: output of ``map_years``.
+        """
         raise NotImplementedError
 
-    def get_test_indices(self, strategy, params):  # noqa
-        """Return indices for test data indices using given strategy."""
+    def get_traintest(self) -> pd.DataFrame:
+        """Shorthand for getting both train and test indices.
+
+        The user will get a flat intervals list similar to the output of `map_years`
+        or `map_data`, but with extra columns containing `train` and `test` labels.
+
+        Returns:
+            Pandas DataFrame with columns specifying whether the interval is
+                part of the train or test datasets.
+
+        Example:
+
+            >>> import s2s.time
+            >>> calendar = s2s.time.AdventCalendar(anchor_date=(10, 15), freq='180d')
+            >>> calendar.map_years(2020, 2021, flat=True)
+            0    (2021-04-18, 2021-10-15]
+            1    (2020-10-20, 2021-04-18]
+            2    (2020-04-18, 2020-10-15]
+            3    (2019-10-21, 2020-04-18]
+            dtype: interval
+            
+            >>> calendar.set_traintest_method("kfold", n_splits = 2)
+            >>> calendar.get_traintest()
+               anchor_year  i_intervals                 intervals fold_0 fold_1
+            0         2021            0  (2021-04-18, 2021-10-15]  train   test
+            1         2021            1  (2020-10-20, 2021-04-18]  train   test
+            2         2020            0  (2020-04-18, 2020-10-15]   test  train
+            3         2020            1  (2019-10-21, 2020-04-18]   test  train
+        """
+        # checker if generated intervals are flat.
+        if self.intervals.ndim != 1:
+            raise ValueError("Please set `flat = True` when calling `map_years` or `map_data`")
+
+        # infer anchor years from give intervals
+        anchor_years = [timestamp.right.year for timestamp in self.intervals[::self._n_intervals]]
+        # Create DataFrame from intervals Series to store the train/test groups
+        traintest_base = pd.DataFrame(data = {
+        'anchor_year': np.repeat(anchor_years, self._n_intervals),
+        'i_intervals': np.tile(range(self._n_intervals), len(anchor_years)),
+        'intervals': self.intervals,
+        })
+
+        # checker if the method is configured.
+        if self._traintest_method is not None:
+            self.traintest = traintest.ALL_METHODS[self._traintest_method](traintest_base,
+                **self._method_kwargs)
+
+        return self.traintest
+
+    def set_traintest_method(self, method: str, **method_kwargs: Optional[dict]):
+        """
+        The user must choose a method here. And the method will be used by
+        all traintest splitting methods.
```

Comment:
"the method will be used by all methods"? This is not accurate.

```suggestion
        Configure the train/test splitting strategy for this calendar instance.
```